Code/Source/RobotImporter/URDF/CollidersMaker.cpp

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "RobotImporter/URDF/CollidersMaker.h"
+#include "RobotImporter/URDF/PrefabMakerUtils.h"
+#include "RobotImporter/URDF/TypeConversions.h"
+#include <AzCore/Serialization/Json/JsonUtils.h>
+#include <AzCore/StringFunc/StringFunc.h>
+#include <AzToolsFramework/API/EditorAssetSystemAPI.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <LmbrCentral/Shape/BoxShapeComponentBus.h>
+#include <LmbrCentral/Shape/CylinderShapeComponentBus.h>
+#include <LmbrCentral/Shape/SphereShapeComponentBus.h>
+#include <SceneAPI/SceneCore/Containers/Scene.h>
+#include <SceneAPI/SceneCore/Containers/SceneManifest.h>
+#include <SceneAPI/SceneCore/Containers/Utilities/Filters.h>
+#include <SceneAPI/SceneCore/DataTypes/Groups/ISceneNodeGroup.h>
+#include <SceneAPI/SceneCore/Events/AssetImportRequest.h>
+#include <SceneAPI/SceneCore/Events/SceneSerializationBus.h>
+#include <SceneAPI/SceneCore/Utilities/SceneGraphSelector.h>
+#include <Source/EditorColliderComponent.h>
+#include <Source/EditorShapeColliderComponent.h>
+
+namespace ROS2
+{
+    CollidersMaker::CollidersMaker(AZStd::string modelPath)
+        : m_modelPath(AZStd::move(modelPath))
+        , m_stopBuildFlag(false)
+    {
+    }
+
+    CollidersMaker::~CollidersMaker()
+    {
+        m_stopBuildFlag = true;
+        m_buildThread.join();
+    };
+
+    void CollidersMaker::BuildColliders(urdf::LinkSharedPtr link)
+    {
+        for (auto collider : link->collision_array)
+        {
+            BuildCollider(collider);
+        }
+
+        if (link->collision_array.empty())
+        {
+            BuildCollider(link->collision);
+        }
+    }
+
+    void CollidersMaker::BuildCollider(urdf::CollisionSharedPtr collision)
+    {
+        if (!collision)
+        { // it is ok not to have collision in a link
+            return;
+        }
+
+        auto geometry = collision->geometry;
+        bool isPrimitiveShape = geometry->type != urdf::Geometry::MESH;
+        if (!isPrimitiveShape)
+        {
+            auto meshGeometry = std::dynamic_pointer_cast<urdf::Mesh>(geometry);
+            auto azMeshPath = GetFullURDFMeshPath(AZ::IO::Path(m_modelPath), AZ::IO::Path(meshGeometry->filename.c_str()));
+
+            AZStd::shared_ptr<AZ::SceneAPI::Containers::Scene> scene;
+            AZ::SceneAPI::Events::SceneSerializationBus::BroadcastResult(
+                scene, &AZ::SceneAPI::Events::SceneSerialization::LoadScene, azMeshPath.c_str(), AZ::Uuid::CreateNull());
+            if (!scene)
+            {
+                AZ_Error("CollisionMaker", false, "Error loading collider. Invalid scene.");
```

Comment:
Is there any ID or name in the collision object that can be logged here as part of the message?